src/adapters/source-adapters/ChainlinkSourceAdapter.sol

```diff
@@ -94,7 +94,9 @@ abstract contract ChainlinkSourceAdapter is DiamondRootOval {
             _searchRoundDataAt(timestamp, roundId, maxTraversal);
 
         // Validate returned data. If it is uninitialized we fallback to returning the current latest round data.
-        if (historicalUpdatedAt > 0) return (historicalAnswer, historicalUpdatedAt, historicalRoundId);
+        if (historicalUpdatedAt > block.timestamp - maxAge()) {
```

Comment:
Question: Can we leave the `historicalUpdatedAt > 0` in this particular source adapter for gas savings given that CL has history?

Reply:
I think it's ideal for maxAge to be respected regardless of the source. It's a lot easier to reason about the system as a whole in that case. It should add only minor additional gas, as maxAge will typically be an immutable variable.